src/core/sys/posix/signal.d

```diff
@@ -710,23 +710,47 @@ else version (Solaris)
 }
 else version( CRuntime_UClibc )
 {
-    struct sigaction_t
+    version ( MIPS32 )
     {
-        static if( true /* __USE_POSIX199309 */ )
+        struct sigaction_t
         {
-            union
+            uint     sa_flags;
+            static if( true /* __USE_POSIX199309 */ )
+            {
+                union
+                {
+                    sigfn_t     sa_handler;
+                    sigactfn_t  sa_sigaction;
+                }
+            }
+            else
             {
                 sigfn_t     sa_handler;
-                sigactfn_t  sa_sigaction;
             }
+            sigset_t    sa_mask;
+            void function() sa_restorer;
         }
-        else
+    }
+    else
+    {
+        struct sigaction_t
```

Comment:
Preferably, if there's a type definition that differs between architectures, then each supported architecture should be explicitly defined.

The else branch then becomes a `static assert(false)` that is intended to alert porters of something that needs checking.

Reply:
uClibc has this common definition, ARM and x86 uses it, mips doesn't (probably others, haven't checked). I guess I can add a "sigaction_common" version for ARM and x86 and then separate the MIPS and add the static assert in the end.